frontend/src/components/SearchTool/SearchTool.js

```diff
@@ -0,0 +1,95 @@
+import Select, { components } from 'react-select';
+import { RowInOptions, StyledOptions, Title } from './SearchTool.styled';
+import { FaSearch } from 'react-icons/fa';
+import { useQuery } from 'react-query';
+import { FaCheck } from 'react-icons/fa';
+import avatar1 from '../../mocks/images/avatars/sample-avatar1.jpg';
+import { UserAvatar } from '../UserAvatar/UserAvatar';
+
+export const SearchTool = (callback) => {
+  const {
+    data: dataAvailableForUserBoards,
+    isLoadingAvailableBoards,
+    isErrorAvailableBoards,
+  } = useQuery('AvailableBoardsSearchTool', async () => {
+    return await fetch('/board?isNearUser=true').then((response) =>
+      response.json().then((data) =>
+        data.boards.map((board) => {
+          return { id: board.id, value: board.boardName, label: board.boardName };
+        }),
+      ),
+    );
+  });
+
+  const {
+    data: dataBoardsUserAlreadyAssigned,
+    isLoadingBoardsAlreadyAssigned,
+    isErrorBoardsAlreadyAssigned,
+  } = useQuery('UserBoards', async () => {
+    return await fetch('/user/1/boards').then((response) =>
+      response.json().then((data) =>
+        data.boards.map((board) => {
+          return { id: board.id, value: board.boardName, label: board.boardName };
+        }),
+      ),
+    );
+  });
+
+  const options = [
+    { value: '1', label: 'temp1', icon: <FaCheck /> },
+    { value: '2', label: 'temp2', icon: <FaCheck /> },
+    { value: '3', label: 'temp3', icon: '' },
+  ];
+
+  const DropdownIndicator = (props) => {
+    return (
+      <components.DropdownIndicator {...props}>
+        <FaSearch />
+      </components.DropdownIndicator>
+    );
+  };
+  const { Option } = components;
+  const IconOption = (props) => (
+    <Option {...props}>
+      <RowInOptions>
+        <UserAvatar userAvatarImage={avatar1} />
+        {props.data.label}
+        <FaCheck />
+      </RowInOptions>
+    </Option>
+  );
```

Comment:
to są jakieś third-party komponenty, dobrze rozumiem ?

Reply:
Możesz elementy z react-select indywidualnie stylować.